examples/kelvin_helmholtz_instability.jl

```diff
@@ -0,0 +1,308 @@
+# # Stratified Kelvin-Helmholtz instability
+#
+# We simulate Kelvin-Helmholtz instability in two-dimensions in ``x, z``,
+
+using Oceananigans
+
+grid = RegularCartesianGrid(size=(64, 1, 65), x=(-5, 5), y=(0, 1), z=(-5, 5),
+                                  topology=(Periodic, Periodic, Bounded))
+
+# # The basic state
+#
+# We're simulating the instability of a sheared and stably-stratified basic state
+# ``U(z)`` and ``B(z)``. Two parameters define our basic state: the Richardson number
+# defined as,
+#
+# ```math
+# Ri = \frac{∂_z B}{∂_z U^2}
+# ```
+#
+# and the width of the stratification layer, ``h``.
+
+using Oceananigans.Fields
+
+shear_flow(x, y, z, t) = tanh(z)
+
+stratification(x, y, z, t, p) = p.h * p.Ri * tanh(z / p.h)
+
+U = BackgroundField(shear_flow)
+B = BackgroundField(stratification, parameters=(Ri=0.1, h=1/4))
+
+# Our basic state thus has a thin layer of stratification in the center of
+# the channel, embedded within a thicker shear layer surrounded by unstratified fluid.
+
+using Plots, Oceananigans.Grids
+
+z = znodes(Cell, grid)
+
+Ri, h = B.parameters
+
+kwargs = (ylabel="z", linewidth=2, label=nothing)
+
+ U_plot = plot(shear_flow.(0, 0, z, 0), z; xlabel="U(z)", kwargs...)
+ B_plot = plot([stratification(0, 0, z, 0, (Ri=Ri, h=h)) for z in z], z; xlabel="B(z)", color=:red, kwargs...)
+Ri_plot = plot(Ri * sech.(z / h).^2 ./ sech.(z).^2, z; xlabel="Ri(z)", color=:black, kwargs...)
+
+plot(U_plot, B_plot, Ri_plot, layout=(1, 3), size=(850, 400))
+
+# # The model
+
+using Oceananigans.Advection
+
+model = IncompressibleModel(timestepper = :RungeKutta3, 
+                              advection = UpwindBiasedFifthOrder(),
+                                   grid = grid,
+                               coriolis = nothing,
+                      background_fields = (u=U, b=B),
+                                closure = IsotropicDiffusivity(ν=5e-5, κ=5e-5),
+                               buoyancy = BuoyancyTracer(),
+                                tracers = :b)
+
+# # A _Power_ful algorithm
+#
+# We will find the most unstable mode of the Kelvin-Helmholtz instability using the "power method". 
+# In brief, if a linear operator ``\mathcal{L}`` has eigenmodes ``u_j`` and corresponding 
```

Comment:
```suggestion
# Linear instabilities such as the Kelvin-Helmholtz instability, are described by equations of the form
# 
# ```math
# ∂_t Φ + L Φ ≈ 0
# ```
# where in our case ``Φ = (u, v, w, b)`` is a vector of the velocities ``u, v, w`` and buoyancy ``b`` and ``L``
# is a linear operator that depends on the base state (the `background_fields`) and other aspects of the problem.

# If ```L`` has eigenmodes ``u_j`` and corresponding 
```

(is it okay to use regular ``L``? It's easier to read the bare comments without the mathcal decoration.

Reply:
I condemn you though for writing ≈ in your suggestion above.
The equality is strict there!
But I'll move Lφ to the right-hand-side... :)